client/client.go

```diff
@@ -431,14 +437,24 @@ func (c *Client) Do(req *http.Request) (*container.Container, *http.Response, er
 		// Check 2xx status codes
 		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
 			obj, err := container.ParseJSON(bodyBytes)
-			if err == nil {
-				log.Printf("[DEBUG] Exit from do method")
-				return obj, resp, err
+			if err != nil {
+				// Attempt retry if JSON parsing fails but status code is 2xx
+				// Assumption here is that packets were somehow corrupted/lost during transmission
+				log.Printf("[ERROR] Error occurred while JSON parsing (2xx status): %+v", err)
+				retry = true
+			} else {
+				// JSON parsing was successful for a 2xx response.
+				// Now, check the custom retry function.
+				if retryFunc != nil && retryFunc(obj) {
+					log.Printf("[DEBUG] Custom retry function indicated a retry is needed for 2xx response")
+					retry = true
```

Comment:
how do we ensure infinite retry mechanism occurs?

Reply:
You mean infinite retry does not occur? At the moment we have a default of max retries set to 2 and number of retries are dependant on max retries at the provider level. If a user still faces issues with deploy they would have to increase max retries at the provider level.

Reply:
Ok assume we will document this behaviour in the resource?